Dockerfile

```diff
@@ -0,0 +1,59 @@
+FROM ubuntu:14.04
+MAINTAINER Kimmo Ahokas <kimmo.ahokas@futurice.com>
+
+# Configure apt to automatically select mirror
+RUN echo "deb mirror://mirrors.ubuntu.com/mirrors.txt trusty main restricted universe\n\
+deb mirror://mirrors.ubuntu.com/mirrors.txt trusty-updates main restricted universe\n\
+deb mirror://mirrors.ubuntu.com/mirrors.txt trusty-security main restricted universe" > /etc/apt/sources.list
+
+# Install packages
+RUN apt-get update && apt-get install -y \
+    git \
+    mongodb \
+    nodejs \
+    npm \
+    supervisor
+
+# Hack to make npm and other nodejs stuff to work properly
+RUN ln -s /usr/bin/nodejs /usr/bin/node
+
+# Set timezone to Europe/Helsinki
+RUN echo 'Europe/Helsinki' > /etc/timezone
+RUN rm /etc/localtime
+RUN ln -s /usr/share/zoneinfo/Europe/Helsinki /etc/localtime
+
+# Set the locale
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+COPY ./docker/supervisord.conf /etc/supervisor/supervisord.conf
+
+# Run the node app as user "lobby"
+RUN useradd -m -s /bin/bash lobby
+RUN mkdir /opt/lobby
+RUN chown -R lobby:lobby /opt/lobby
+WORKDIR /opt/lobby
+
+# install global dependencies
+RUN npm install -g bower sails@0.10.3 grunt-cli
+
+# Change user to lobby and install rest
+USER lobby
+ENV NPM_ENV=production
+COPY package.json bower.json Gruntfile.js .bowerrc .sailsrc /opt/lobby/
+RUN npm install
+
+# Copy the project files
+COPY . /opt/lobby
+
+USER root
+RUN chown -R lobby:lobby /opt/lobby
+
+EXPOSE 8080
+
+# Default startup command
+# Note: supervisord will pass all environment variables to processes
+
+CMD ["/usr/bin/supervisord", "-c", "/etc/supervisor/supervisord.conf"]
```

Comment:
Adding the volume might be a good idea. Or even moving the mongodb to
different container. However, it seems that the app does not really save
anything for long time in that DB, it kind of acts as a cache or something.

In the end the reason for doing this is to run the lobby service on Deis:
https://confluence.futurice.com/display/infra/app.futurice.com+PaaS+with+Deis

On Tue, Sep 15, 2015 at 10:39 AM, Oleg Grenrus <notifications@github.com>
wrote:

> In Dockerfile
> <https://github.com/futurice/lobby/pull/65#discussion_r39482030>:
>
> > +ENV NPM_ENV=production
> > +COPY package.json bower.json Gruntfile.js .bowerrc .sailsrc /opt/lobby/
> > +RUN npm install
> > +
> > +# Copy the project files
> > +COPY . /opt/lobby
> > +
> > +USER root
> > +RUN chown -R lobby:lobby /opt/lobby
> > +
> > +EXPOSE 8080
> > +
> > +# Default startup command
> > +# Note: supervisord will pass all environment variables to processes
> > +
> > +CMD ["/usr/bin/supervisord", "-c", "/etc/supervisor/supervisord.conf"]
>
> I see, you run mongodb in the container too.
>
> Could you have VOLUME <https://docs.docker.com/reference/builder/#volume>
> for the mongodb data?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/futurice/lobby/pull/65/files#r39482030>.
>



-- 
Kimmo Ahokas · +358 50 533 1351
IT Specialist · Futurice Oy
www.futurice.com · twitter.com/futurice